lib/formations.js

```diff
@@ -197,6 +216,43 @@ function formation_payload_check(payload, sizes) {
     'A healthcheck was specified for a non-web based application, the path should not be set.');
   assert.ok(payload.type === 'web' || (payload.type !== 'web' && !payload.removeHealthcheck),
     'Cannot remove healthcheck on non web based dyno.');
+  assert.ok(payload.type === 'web' || payload.oneoff || (payload.type !== 'web' && payload.command !== '' && payload.command),
+    'A run command must be specified for any type outside of a web process.');
+  assert.ok(!payload.oneoff || payload.type !== 'web' || (payload.oneoff && payload.type !== 'web'),
```

Comment:
This seems wrong to me, what if someone creates a `worker` formation type? Wouldn't that get trapped here?

Reply:
AFAIK the only actual special type of named dyno is "web". I don't think we do any checks in the code for "worker", and so it's basically "web" and "non-web" dynos. If someone created a one-off named "worker" then I don't think it should be treated any differently, only when they try and create a one-off "web" should we throw an error.

Thoughts?

Reply:
On second review this is accurate, i'm brain dead today I tells ya.